daemon/network.go

```diff
@@ -32,37 +32,48 @@ func (daemon *Daemon) NetworkControllerEnabled() bool {
 
 // FindNetwork function finds a network for a given string that can represent network name or id
 func (daemon *Daemon) FindNetwork(idName string) (libnetwork.Network, error) {
-	// Find by Name
-	n, err := daemon.GetNetworkByName(idName)
-	if err != nil && !isNoSuchNetworkError(err) {
+	// 1. match via full ID.
+	n, err := daemon.GetNetworkByID(idName)
+	if err == nil {
+		return n, nil
+	}
+	if !isNoSuchNetworkError(err) {
 		return nil, err
 	}
 
-	if n != nil {
+	// 2. match via full name and ID prefix.
+	n, err = daemon.GetNetworkByName(idName)
+	if err == nil {
 		return n, nil
 	}
+	if !isNoSuchNetworkError(err) {
+		return nil, err
+	}
 
-	// Find by id
-	return daemon.GetNetworkByID(idName)
+	// 3. match by ID prefix
+	list := daemon.GetNetworksByIDPrefix(idName)
+	if len(list) == 0 {
+		return nil, libnetwork.ErrNoSuchNetwork(idName)
+	}
+	if len(list) > 1 {
+		return nil, libnetwork.ErrInvalidID(idName)
+	}
+	return list[0], nil
 }
 
 func isNoSuchNetworkError(err error) bool {
 	_, ok := err.(libnetwork.ErrNoSuchNetwork)
 	return ok
 }
 
-// GetNetworkByID function returns a network whose ID begins with the given prefix.
-// It fails with an error if no matching, or more than one matching, networks are found.
-func (daemon *Daemon) GetNetworkByID(partialID string) (libnetwork.Network, error) {
-	list := daemon.GetNetworksByID(partialID)
-
-	if len(list) == 0 {
-		return nil, libnetwork.ErrNoSuchNetwork(partialID)
-	}
-	if len(list) > 1 {
-		return nil, libnetwork.ErrInvalidID(partialID)
+// GetNetworkByID function returns a network whose ID matches the given ID.
+// It fails with an error if no matching network is found.
+func (daemon *Daemon) GetNetworkByID(id string) (libnetwork.Network, error) {
+	c := daemon.netController
+	if c == nil {
+		return nil, libnetwork.ErrNoSuchNetwork(id)
```

Comment:
I see why you have to pass it as ErrNoSuchNetwork, so that it goes to the next check. you said on windows is nil the controller, any idea then how it works the GetNetwork there?